src/main/java/eu/deltasource/internship/hotelmanagementsystem/Room.java

```diff
@@ -0,0 +1,212 @@
+package eu.deltasource.internship.hotelmanagementsystem;
+
+
+/**
+ * Class Room has 5 private members
+ * methods - setters, getters, constructors and other methods
+ */
+
+import eu.deltasource.internship.hotelmanagementsystem.hotel.service.domain.commodities.AbstractCommodity;
+import eu.deltasource.internship.hotelmanagementsystem.hotel.service.domain.commodities.Bed;
+import eu.deltasource.internship.hotelmanagementsystem.hotel.service.domain.commodities.Booking;
+
+
+import java.time.LocalDate;
+import java.util.*;
+
+public class Room {
+
+    private int roomNum;
+    private short capacity;
+    Set<AbstractCommodity> commodities;
+    Set<LocalDate> maintenanceDates;
+    Set<Booking> bookings;
+
+    /**
+     * Parametrized constructor
+     *
+     * @param roomNum          ID of the room
+     * @param commodities      set of commodities
+     * @param maintenanceDates set of
+     * @param bookings         set of bookings
+     */
+
+    public Room(int roomNum, Set<AbstractCommodity> commodities,
+                Set<LocalDate> maintenanceDates, Set<Booking> bookings) {
+        this.roomNum = roomNum;
+        this.commodities = commodities;
+        this.maintenanceDates = maintenanceDates;
+        this.bookings = bookings;
+    }
+
+    /**
+     * Default constructor
+     */
+
+
+    public Room() {
+
+    }
+
+    public int getRoomNum() {
+        return roomNum;
+    }
+
+    public short getCapacity() {
+        return capacity;
+    }
+
+    public Set<AbstractCommodity> getCommodities() {
+        return commodities;
+    }
+
+    public Set<LocalDate> getMaintenanceDates() {
+        return maintenanceDates;
+    }
+
+    public Set<Booking> getBookings() {
+        return bookings;
+    }
+
+    public void setRoomNum(int roomNum) {
+        this.roomNum = roomNum;
+    }
+
+    public void setCapacity(short capacity) {
+        this.capacity = capacity;
+    }
+
+    public void setCommodities(Set<AbstractCommodity> commodities) {
+        this.commodities = commodities;
+    }
+
+    public void setMaintenanceDates(Set<LocalDate> maintenanceDates) {
+        this.maintenanceDates = maintenanceDates;
```

Comment:
We can enhance the immutability of the field by creating a new HashSet with the maintenanceDates. f.e. `this.maintenanceDates = new HashSet(maintenanceDates);`
If we do it this way, the object which uses this method will still be able to modify the set outside the object.
He could do:`room.setMaintenanceDates(someMaintenanceDates);
maintenanceDates.add(new LocalDate(...));`